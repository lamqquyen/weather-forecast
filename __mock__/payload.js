export const locationPayload = [
  {
    title: 'San Francisco',
    location_type: 'City',
    woeid: 2487956,
    latt_long: '37.777119, -122.41964'
  }
]

export const weatherPayload = {
  consolidated_weather: [{
    id: 5863259740569600,
    weather_state_name: 'Heavy Cloud',
    weather_state_abbr: 'hc',
    wind_direction_compass: 'W',
    created: '2020-12-06T06:34:14.572087Z',
    applicable_date: '2020-12-06',
    min_temp: 5.41,
    max_temp: 16.305,
    the_temp: 15.059999999999999,
    wind_speed: 4.547368277541823,
    wind_direction: 276,
    air_pressure: 1024.5,
    humidity: 49,
    visibility: 17.01345392905432,
    predictability: 71,
  }]
}
