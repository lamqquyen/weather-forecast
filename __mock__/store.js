import thunk from 'redux-thunk'
import configureMockStore from 'redux-mock-store' 

const mockStore = configureMockStore([thunk])
const initialState = {
  location: {
    queryResults: [],
    loading: false
  },
  weather: {
    info: {}
  }
}

export {mockStore, initialState}
