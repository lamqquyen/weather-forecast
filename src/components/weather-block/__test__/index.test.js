import React from "react";
import SearchSuggestion from "../index";
import TestRenderer from "react-test-renderer";
import WeatherBlock from '../index'
import {weatherPayload} from '../../../../__mock__/payload'

describe("WeatherBlock", () => {
  it("renders correctly", () => {
    const tree = TestRenderer.create(<WeatherBlock info={weatherPayload}/>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
