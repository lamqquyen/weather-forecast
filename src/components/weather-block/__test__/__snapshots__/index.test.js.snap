// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherBlock renders correctly 1`] = `
<div
  className="weather-block__wrapper"
>
  <div
    className="weather-info__wrapper"
  >
    <div
      className="weather-info__wrapper--left"
      md={6}
    >
      <div
        className="weather-info__text--location"
      >
         
      </div>
      <div
        className="weather-info__text--day"
      >
        Monday, 06 December 2020
      </div>
      <div
        className="weather-info__text--state-name"
      >
        Heavy Cloud
      </div>
    </div>
    <div
      className="weather-info__wrapper--right"
      md={6}
    >
      <div
        className="weather-info__info-wrapper"
      >
        <div>
          Humidity: 
          49
          %
        </div>
        <div>
          Wind: 
          4.55
           mph
        </div>
        <div>
          Air Pressure: 
          1024.5
           mbar
        </div>
        <div>
          Visibility: 
          17.01
           miles
        </div>
      </div>
      <div>
        <img
          className="weather-info__icon"
          src="https://www.metaweather.com/static/img/weather/hc.svg"
        />
        <div
          className="weather-info__text--temperature"
        >
          16
          °C
        </div>
      </div>
    </div>
  </div>
  <div
    className="weather-block__cards-wrapper"
  >
    <div
      className="weather-card__wrapper weather-card__wrapper--selected"
      onClick={[Function]}
    >
      <div
        className="weather-card__text--day"
      >
        MON
      </div>
      <div
        className="weather-card__text--date"
      >
        12 - 06
      </div>
      <img
        className="weather-card__icon"
        src="https://www.metaweather.com/static/img/weather/hc.svg"
      />
      <div
        className="weather-card__temperature-wrapper"
      >
        <div
          className="weather-card__text--max-temperature"
        >
          16
          °
        </div>
        <div
          className="weather-card__text--min-temperature"
        >
          5
          °
        </div>
      </div>
    </div>
  </div>
</div>
`;
